test/org/jetbrains/plugins/scala/codeInspection/collections/IfElseToFilteredOptionInspectionTest.scala

```diff
@@ -0,0 +1,46 @@
+package org.jetbrains.plugins.scala.codeInspection.collections
+
+import com.intellij.testFramework.EditorTestUtil
+
+/**
+  * @author mattfowler
+  */
+class IfElseToFilteredOptionInspectionTest extends OperationsOnCollectionInspectionTest {
+  override protected val classOfInspection: Class[_ <: OperationOnCollectionInspection] =
+    classOf[IfElseToFilteredOptionInspection]
+
+  override protected val hint: String = "Replace if with filtered option"
+
+  import EditorTestUtil.{SELECTION_END_TAG => END, SELECTION_START_TAG => START}
+
+  private val evenFunction = "def isEven(x:Int) = x % 2 == 0;"
+
+  def testShouldReplaceWhenReturningSome() = {
+    doTest(
+      s"$evenFunction ${START}if (isEven(2)) Some(2) else None$END",
+      s"$evenFunction if (isEven(2)) Some(2) else None",
+      s"$evenFunction Some(2).filter(isEven)"
+    )
+  }
+
+  def testShouldReplaceWhenReturningOption() = {
+    doTest(
+      s"$evenFunction ${START}if (isEven(2)) Option(2) else None$END",
+      s"$evenFunction if (isEven(2)) Option(2) else None",
+      s"$evenFunction Option(2).filter(isEven)"
+    )
+  }
+
+  def testShouldReplaceWithMethodCallAsParam() = {
+    val getInt = "def getInt() = 2;"
+    doTest(
+      s"$evenFunction $getInt ${START}if (isEven(getInt())) Option(getInt()) else None$END",
```

Comment:
We need to be sure that result expression is equivalent to the old one. It is not the case if argument has side effects, because it is computed once instead of 2 times. We have org.jetbrains.plugins.scala.util.SideEffectsUtil.hasNoSideEffects for such cases.